data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/PipelineParser.java

```diff
@@ -306,4 +314,18 @@ String getName() {
             return name;
         }
     }
+
+    Duration getPeerForwarderDrainTimeout(final DataPrepperConfiguration dataPrepperConfiguration) {
+        return Optional.ofNullable(dataPrepperConfiguration)
+                .map(DataPrepperConfiguration::getPeerForwarderConfiguration)
+                .map(PeerForwarderConfiguration::getDrainTimeout)
+                .orElse(Duration.ofSeconds(0));
+    }
+
+    List<Buffer> getSecondaryBuffers() {
```

Comment:
This could be a useful method in the `PeerForwarderProvider` directly. That would make it more testable.